sqlmesh/core/engine_adapter/databricks.py

```diff
@@ -48,9 +48,13 @@ class DatabricksEngineAdapter(SparkEngineAdapter):
         },
     )
 
-    def __init__(self, *args: t.Any, **kwargs: t.Any):
-        super().__init__(*args, **kwargs)
+    def __init__(self, *args: t.Any, **kwargs: t.Any) -> None:
+        super().__init__(*args, **{**kwargs, "skip_connection_post_init": True})
         self._set_spark_engine_adapter_if_needed()
+        if not kwargs.get("skip_connection_post_init"):
```

Comment:
Why not just let the base class call `_connection_post_init`?

Reply:
Because the spark engine adapter would not be set by that time.